From ee8cdc7b141767c59c03494404e3eb5885642175 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Mon, 23 Jun 2014 21:49:27 -0400
Subject: [fs] nfsd: clean up unnecessary temporary variable in nfsd4_decode_fattr

Message-id: <1403560314-32525-26-git-send-email-bfields@redhat.com>
Patchwork-id: 84526
O-Subject: [Patch RHEL7 025/172] nfsd: clean up unnecessary temporary variable in nfsd4_decode_fattr
Bugzilla: 1111719
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Kinglong Mee <kinglongmee@gmail.com>

host_err was only used for nfs4_acl_new.
This patch delete it, and return nfserr_jukebox directly.

Signed-off-by: Kinglong Mee <kinglongmee@gmail.com>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: eba1c99ce4590506516ec801d991e36aa8b0d436
Bugzilla: 1111719
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4xdr.c b/fs/nfsd/nfs4xdr.c
index 776d2f6..b77f6bd 100644
--- a/fs/nfsd/nfs4xdr.c
+++ b/fs/nfsd/nfs4xdr.c
@@ -276,7 +276,6 @@ nfsd4_decode_fattr(struct nfsd4_compoundargs *argp, u32 *bmval,
  int expected_len, len = 0;
  u32 dummy32;
  char *buf;
- int host_err;
 
  DECODE_HEAD;
  iattr->ia_valid = 0;
@@ -303,10 +302,9 @@ nfsd4_decode_fattr(struct nfsd4_compoundargs *argp, u32 *bmval,
    return nfserr_resource;
 
   *acl = nfs4_acl_new(nace);
-  if (*acl == NULL) {
-   host_err = -ENOMEM;
-   goto out_nfserr;
-  }
+  if (*acl == NULL)
+   return nfserr_jukebox;
+
   defer_free(argp, kfree, *acl);
 
   (*acl)->naces = nace;
@@ -444,10 +442,6 @@ nfsd4_decode_fattr(struct nfsd4_compoundargs *argp, u32 *bmval,
   goto xdr_error;
 
  DECODE_TAIL;
-
-out_nfserr:
- status = nfserrno(host_err);
- goto out;
 }
 
 static __be32
-- 
1.7.1